src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -1972,9 +1972,35 @@ private void CompileAsVoidRemoveRethrow(Expression expr)
             CompileBlockEnd(end);
         }
 
+        private sealed class TryVisitor : ExpressionVisitor
+        {
+            public static readonly TryVisitor Instance;
+            
+            static TryVisitor()
+            {
+                Instance = new TryVisitor();
+            }
+
+            private static Expression ForceVoid(Expression node)
+            {
+                return node.Type == typeof(void) ? node : Expression.Block(typeof(void), node);
+            }
+
+            protected internal override Expression VisitTry(TryExpression node)
+            {
+                return node.Type != typeof(void) ? node : node.Update(ForceVoid(node.Body), Visit(node.Handlers, VisitCatchBlock), node.Finally, node.Fault);
+            }
+
+            protected override CatchBlock VisitCatchBlock(CatchBlock node)
+            {
+                var body = node.Body;
+                return body.Type == typeof(void) ? node : node.Update(node.Variable, node.Filter, ForceVoid(body));
+            }
+        }
+
         private void CompileTryExpression(Expression expr)
         {
-            var node = (TryExpression)expr;
+            var node = (TryExpression)TryVisitor.Instance.Visit(expr);
```

Comment:
No, I don't need to handle finally and fault for those reasons, they're already going to not be visited. I do still need to double check they don't have similar issues when I'm on a computer rather than a phone.